log4j-layout-template-json/src/main/java/org/apache/logging/log4j/layout/template/json/resolver/TimestampResolver.java

```diff
@@ -331,119 +273,48 @@ private static EventResolver createEpochResolver(final TemplateResolverConfig co
         throw new IllegalArgumentException("invalid epoch configuration: " + config);
     }
 
-    private static final class EpochResolutionRecord {
-
-        private static final int MAX_LONG_LENGTH =
-                String.valueOf(Long.MAX_VALUE).length();
-
-        private final MutableInstant instant = new MutableInstant();
-
-        private final char[] resolution =
-                new char[ /* integral: */MAX_LONG_LENGTH + /* dot: */ 1 + /* fractional: */ MAX_LONG_LENGTH];
-
-        private int resolutionLength;
-
-        private EpochResolutionRecord() {
-            instant.initFromEpochSecond(-1, 0);
-        }
-    }
-
-    private abstract static class EpochResolver implements EventResolver {
-
-        private final Lock lock = new ReentrantLock();
-
-        private final EpochResolutionRecord resolutionRecord = new EpochResolutionRecord();
-
-        @Override
-        public void resolve(final LogEvent logEvent, final JsonWriter jsonWriter) {
-            lock.lock();
-            try {
-                unsynchronizedResolve(logEvent, jsonWriter);
-            } finally {
-                lock.unlock();
-            }
-        }
-
-        private void unsynchronizedResolve(final LogEvent logEvent, final JsonWriter jsonWriter) {
-            final Instant logEventInstant = logEvent.getInstant();
-            if (logEventInstant.equals(resolutionRecord.instant)) {
-                jsonWriter.writeRawString(resolutionRecord.resolution, 0, resolutionRecord.resolutionLength);
-            } else {
-                resolutionRecord.instant.initFrom(logEventInstant);
-                final StringBuilder stringBuilder = jsonWriter.getStringBuilder();
-                final int startIndex = stringBuilder.length();
-                resolve(logEventInstant, jsonWriter);
-                resolutionRecord.resolutionLength = stringBuilder.length() - startIndex;
-                stringBuilder.getChars(startIndex, stringBuilder.length(), resolutionRecord.resolution, 0);
-            }
-        }
-
-        abstract void resolve(Instant logEventInstant, JsonWriter jsonWriter);
-    }
-
-    private static final EventResolver EPOCH_NANOS_RESOLVER = new EpochResolver() {
-        @Override
-        void resolve(final Instant logEventInstant, final JsonWriter jsonWriter) {
-            final long nanos = epochNanos(logEventInstant);
-            jsonWriter.writeNumber(nanos);
-        }
+    private static final EventResolver EPOCH_NANOS_RESOLVER = (logEvent, jsonWriter) -> {
+        final StringBuilder buffer = jsonWriter.getStringBuilder();
+        final Instant instant = logEvent.getInstant();
+        InstantNumberFormatter.EPOCH_NANOS.formatTo(buffer, instant);
```

Comment:
Is this behavioral change intentional? The type of the field is changed from Number to String.

Reply:
The output is still a number, though this time directly written to the SB. Am I missing something?

Reply:
Ah, so you are writing a **raw** string to JSON? I didn't know this was possible.